From: Matthew Wilcox <matthew.r.wilcox@intel.com>
Date: Sat, 19 Mar 2011 14:55:38 -0400
Subject: NVMe: Add compat_ioctl
Git-commit: 4948168280b269a514045766ddd872cfac5968e1
References: FATE#313627
Patch-Mainline: v3.7-rc1

Make ioctls work for 32-bit applications on 64-bit kernels.  The structures
are defined to be the same for both 32- and 64-bit applications, so
we can use the same handler for both.

Reported-by: Arnd Bergmann <arnd@arndb.de>
Signed-off-by: Matthew Wilcox <matthew.r.wilcox@intel.com>
Acked-by: Hannes Reinecke <hare@suse.de>
---
 drivers/block/nvme.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/drivers/block/nvme.c b/drivers/block/nvme.c
index f94f173..d0b5262 100644
--- a/drivers/block/nvme.c
+++ b/drivers/block/nvme.c
@@ -1146,6 +1146,7 @@ static int nvme_ioctl(struct block_device *bdev, fmode_t mode, unsigned int cmd,
 static const struct block_device_operations nvme_fops = {
 	.owner		= THIS_MODULE,
 	.ioctl		= nvme_ioctl,
+	.compat_ioctl	= nvme_ioctl,
 };
 
 static void nvme_resubmit_bios(struct nvme_queue *nvmeq)
-- 
1.7.4.2

